Functions/Gherkin.ps1

```diff
@@ -1,5 +1,9 @@
 # Work around bug in PowerShell 2 type loading...
-Microsoft.PowerShell.Core\Import-Module -Name "${Script:PesterRoot}\lib\Gherkin.dll"
+if(!$IsCoreClr) {
+    Microsoft.PowerShell.Core\Import-Module -Name "${Script:PesterRoot}\lib\core\Gherkin.dll"
+} else {
+    Microsoft.PowerShell.Core\Import-Module -Name "${Script:PesterRoot}\lib\legacy\Gherkin.dll"
```

Comment:
The file `lib\legacy\gherkin.dll` has the first letter small and I don't like that the first letter for `lib\core\Gherkin.dll` is not small too.

Additionally, to achieve compatibility with Linux and macOS we need use code like below
```
if(!$IsCoreCLR) {
    Microsoft.PowerShell.Core\Import-Module -Name "${Script:PesterRoot}\lib\legacy\gherkin.dll"
} else {
    [String]$GherkinCoreDllPath = "${Script:PesterRoot}{0}lib{0}core{0}Gherkin.dll" -f [System.IO.Path]::DirectorySeparatorChar
    Microsoft.PowerShell.Core\Import-Module -Name $GherkinCoreDllPath
}
```